homeassistant/components/sma/config_flow.py

```diff
@@ -0,0 +1,226 @@
+"""Config flow for the sma integration."""
+import logging
+
+import aiohttp
+import pysma
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PATH,
+    CONF_SENSORS,
+    CONF_SSL,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_CUSTOM,
+    CONF_FACTOR,
+    CONF_GROUP,
+    CONF_KEY,
+    CONF_UNIT,
+    DEVICE_INFO,
+    GROUPS,
+)
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: core.HomeAssistant, data: dict):
+    """Validate the user input allows us to connect."""
+    session = async_get_clientsession(hass, verify_ssl=data[CONF_VERIFY_SSL])
+
+    protocol = "https" if data[CONF_SSL] else "http"
+    url = f"{protocol}://{data[CONF_HOST]}"
+
+    # This will also test device connectivity. Raises ClientError on failure
+    res = await session.get(f"{url}/data/l10n/en-US.json")
```

Comment:
This is also too device specific and should rather be in the library. But why can't we just rely on establishing a successful connection with `new_session()`?

Reply:
I will move this to the pysma library. The get is needed as the SMA device will return the manufacturer and device type as an ID. The translation file is used to lookup the ID and get the actual string.